universal-login-commons/src/core/utils/InitializationHandler.ts

```diff
@@ -0,0 +1,56 @@
+type State<T, B> = {
+  kind: 'None';
+} | {
+  kind: 'Initializing';
+  value: Promise<T>;
+} | {
+  kind: 'Initialized';
+  value: Promise<T>;
```

Comment:
```suggestion
  value: T;
```
we have already awaited the value at this point